components/http-accept.js

```diff
@@ -1,55 +1,59 @@
 // http-accept.js
 
 var _ = require('underscore');
-var body = require('body');
-var noflo = require('noflo');
+var getRawBody = require('raw-body');
+var typer = require('media-typer');
 
 var basenode = require('./base-node');
+var promiseComponent = require('./promise-component');
 
-exports.getComponent = function() {
-    return _.extend(new noflo.Component({
-        outPorts: {
-            out: {
-                description: "Request body as a String",
-                datatype: 'string'
-            },
-            rejected: {
-                description: "HTTP 415 Unsupported Media Type request/response pair {req, res}",
-                datatype: 'object'
-            },
-            accepted: {
-                description: "HTTP 202 Accepted request/response pair {req, res}",
-                datatype: 'object'
-            }
+exports.getComponent = promiseComponent({
+    description: "Extracts the request body as a String and produces a 202 Accepted response. Optionally, also filters on request content type.",
+    icon: 'sign-in',
```

Comment:
I thought the arrow going into a box was an appropriate imagery for this (nothing to do with the action of signing-in though).